### PR TITLE
Add balance metrics to op-propose & op-batcher

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This adds a utility function to the `op-service` package which launches a go rountine
to check the balance of a given account. This is helpful for in process monitoring of
the balance of an account for a specific service (like the batcher or proposer).

**Tests**

Tested locally with devnet.

**Metadata**
- Fixes ENG-2713

### CODE REVIEW

File: op-service/metrics/balance.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// 10^18
+var etherScalar = new(big.Int).Exp(big.NewInt(10), big.NewInt(18), nil)
```
Comment (tynes) [inline op-service/metrics/balance.go:16]: This constant is defined in many places throughout the monorepo and also in geth itself

Comment (trianglesphere) [reply op-service/metrics/balance.go:16]: Do you know the names of the other definitions? There's one with the same name (also private) in the op-e2e package. In geth there is a private one in `cmd/faucet`.

**Review by mslipper — APPROVED** [review]

File: op-service/metrics/balance.go (hunk 1 above)
Comment (protolambda) [reply op-service/metrics/balance.go:16]: e2eutils or action testing has one too iirc. We should probably just put a global in the `op-node/eth` package

Comment (tynes) [reply op-service/metrics/balance.go:16]: https://github.com/ethereum/go-ethereum/blob/b628d7276624c2d8ea7dd97d2259a2c2fce7d3cc/params/denomination.go

**Review by protolambda — APPROVED** [review]

File: op-service/metrics/balance.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+// Cancel the supplied context to shut down the go routine
+func LaunchBalanceMetrics(ctx context.Context, log log.Logger, r *prometheus.Registry, ns string, client *ethclient.Client, account common.Address) {
+	go func() {
+		balanceGuage := promauto.With(r).NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "balance",
+			Help:      "balance (in ether) of account " + account.String(),
+		})
+
+		ticker := time.NewTicker(10 * time.Second)
```
Comment (tynes) [inline op-service/metrics/balance.go:38]: I feel like 10 seconds is a bit fast, I feel like 1 minute might be plenty, its just a lot of requests. The cost of running a node decreases the amount of users running nodes

Comment (trianglesphere) [reply op-service/metrics/balance.go:38]: will do. I don't think we actually need that fine grain data b/c we can set alerts based on projected balances (using a linear prediction in grafana).

File: op-service/metrics/balance.go (hunk 1 above)
Comment (trianglesphere) [reply op-service/metrics/balance.go:16]: Oh I see, it's 1E18 there. I'm probably gonna use that + `big.NewRat` rather than going through `big.Ints` here.

**Comment by tynes** [comment]
Will approve pending changes implemented

**Pull request merged**